File: app/src/main/java/com/simplecity/amp_library/dagger/component/AppComponent.java
package com.simplecity.amp_library.dagger.component;

import com.simplecity.amp_library.dagger.module.ActivityModule;
import com.simplecity.amp_library.dagger.module.AppModule;
import com.simplecity.amp_library.dagger.module.DrawerModule;
import com.simplecity.amp_library.dagger.module.ModelsModule;
import com.simplecity.amp_library.dagger.module.MultiSheetModule;
import dagger.Component;
import javax.inject.Singleton;

@Singleton
@Component(modules = {
        AppModule.class,
        ModelsModule.class,
        MultiSheetModule.class,
        DrawerModule.class
})

public interface AppComponent {

    ActivityComponent plus(ActivityModule module);
}

File: app/src/main/java/com/simplecity/amp_library/di/app/AppComponent.kt
package com.simplecity.amp_library.di.app

import com.simplecity.amp_library.ShuttleApplication
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(
    modules = [
        AndroidSupportInjectionModule::class,
        AppModule::class,
        AppAssistedModule::class,
        RepositoryModule::class
    ]
)
interface AppComponent : AndroidInjector<ShuttleApplication> {
    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<ShuttleApplication>()
}